apps/dot_ext/tests/test_api_error_codes.py

```diff
@@ -0,0 +1,184 @@
+from apps.test import BaseApiTest
+from http import HTTPStatus
+import json
+import pytz
+from random import randint
+from apps.authorization.models import (
+    DataAccessGrant,
+)
+from apps.constants import AccessType
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+from unittest import mock
+
+
+class StubDate(datetime):
+    pass
+
+
+# These tests cover `validate_app_is_active` in `utils.py`
+# In particular, they cover whether or not we get the right error
+# codes back for a given call/situation.
+class TestDataAccessPermissions(BaseApiTest):
+
+    # In this test, we should get back a `HTTPStatus.UNAUTHORIZED` when we try
+    # and refresh the token on a ONE_TIME application.
+    def test_one_time_access_unauthorized_to_refresh(self):
+        refresh_responses = [
+            {
+                "app_data_access_type": AccessType.ONE_TIME,
+                "status_code": HTTPStatus.FORBIDDEN,
+                "expected_in": "User data access grant expired"
```

Comment:
I'm not seeing this `expected_in` being used, we should update the function calls below, or remove the field from here. I also thought that the one-time case had a different message than this to say that refresh is not allowed, so if we do change this to test for the message text, we should double check that too.